sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/DataSourceReader.java

```diff
@@ -76,5 +76,5 @@
    * If this method fails (by throwing an exception), the action would fail and no Spark job was
    * submitted.
    */
-  List<DataReaderFactory<Row>> createDataReaderFactories();
+  List<InputPartition<Row>> planInputPartitions();
```

Comment:
I think plan is a more accurate verb. To some Java people, `get` implies that the call is very cheap because it is associated with getters, which typically just return a field's value. Since that's not the case here and callers shouldn't consider this method cheap, I think it makes sense to use a different name that reflects what is actually happening: split planning.